realsense2_camera/include/realsense_node_factory.h

```diff
@@ -66,10 +67,15 @@ namespace realsense2_camera
         virtual void onInit() override;
         void tryGetLogSeverity(rs2_log_severity& severity) const;
 
+        tf2_ros::Buffer _tf_buffer;
+        tf2_ros::TransformListener* _tf_listener;
```

Comment:
Is it necessary that this is a pointer, or can you just have an object as a member such that `new` allocation and calling `delete` is not required?

Reply:
Pointer is necessary because it needs to be initialized with the _tf_buffer object. As class member data, it cannot be initialized otherwise. I will see if we can move the listener to a member function local variable, then we can avoid pointer as well as allocation/deallocation.

Reply:
Checked this: Both _tf_buffer and _tf_listener objects need to be static or class members. Making them static fails the code in the first attempt as it takes some time to initialize. So the sensor init code succeeds only in the second attempt or later. Making them class member and creating the object in the class constructor has the objects ready before the sensor initialization code. This seems to be the best option.

Reply:
Hmm in some other PR, I used a local `tf2_ros::TransformListener` and that seemed to work

Reply:
>  The TransformListener object should be scoped to persist otherwise it's cache will be unable to fill and almost every query will fail. A common method is to make the TransformListener object a member variable of a class.

[From ros wiki](http://wiki.ros.org/tf/Tutorials/Writing%20a%20tf%20listener%20%28C%2B%2B%29) . 
I remember having similar issue and needed to make the listener object persist rather than instantaneous creation/destruction

Reply:
Added more clear comments to explain the design